.github/workflows/maven.yml

```diff
@@ -0,0 +1,62 @@
+# Builds using maven
+# Pass the arguments you wish to run to maven_commands variable, default command will be mvn install
+---
+name: Maven
+
+on: 
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 0 * * 0'
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        java: [1.8, 11]
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    env:
+      maven_commands: install # default is install
+      artifact_dir: target/*
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK ${{ matrix.java }}
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+      - name: Cache Maven packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Build
+        run: mvn ${{ env.maven_commands }}
+      - name: Upload artifacts
+        if: matrix.os == 'ubuntu-latest' && matrix.java == '1.8'
+        uses: actions/upload-artifact@v2
+        with:
+          name: artifacts
+          path: ${{ env.artifact_dir }}
+          if-no-files-found: error
+  release:
+    runs-on: ubuntu-latest
+    needs: build
+    if: startsWith(github.ref, 'refs/tags')
+    env:
+      publishing_target: github # default is github
```

Comment:
From the template perspective, are you foreseeing jobs publishing to multiple targets? If so, will this need to become an array?